src/Language/ASKEE/Core/Expr.hs

```diff
@@ -325,3 +326,37 @@ orderDecls = fst . partOrderDecls Set.empty
 
 
 
+asExpr :: Expr -> Expr.Expr
+asExpr expr =
+  case expr of
+    Literal l -> doLit l
+    Op1 op1 e1 -> doOp1 op1 e1
+    Op2 op2 e1 e2 -> doOp2 op2 e1 e2
+    Var i -> Expr.Var i
+    If p t f -> Expr.If (asExpr p) (asExpr t) (asExpr f)
+    Fail _ -> undefined
```

Comment:
A more descriptive error message is likely warranted here. Why not something like this?

```suggestion
    Fail err -> error err
```